puppet-chrome/lib/Browser.ts

```diff
@@ -139,25 +139,25 @@ export class Browser extends TypedEventEmitter<IBrowserEvents> implements IPuppe
       context?.onSharedWorkerAttached(devtoolsSession, targetInfo).catch(() => null);
     }
 
-    if (event.waitingForDebugger && targetInfo.type === 'service_worker') {
+    if (targetInfo.type === 'service_worker') {
       const devtoolsSession = this.connection.getSession(sessionId);
       if (!devtoolsSession) return;
 
-      const promise = devtoolsSession.send('Runtime.runIfWaitingForDebugger').catch(() => null);
-      if (targetInfo.url.startsWith('chrome-extension://')) {
-        promise
-          .then(() => this.devtoolsSession.send('Target.detachFromTarget', { sessionId }))
-          .catch(() => null);
+      if (event.waitingForDebugger) {
+        devtoolsSession.send('Runtime.runIfWaitingForDebugger').catch(() => null);
       }
+      const context = this.getBrowserContext(targetInfo.browserContextId);
+      context.plugins.onServiceWorkerStarted(devtoolsSession, event).catch(() => null);
     }
 
     if (
       targetInfo.type === 'other' &&
-      targetInfo.url.startsWith('devtools://devtools') &&
-      this.onDevtoolsAttached
+      targetInfo.url.startsWith('devtools://devtools')
     ) {
       const devtoolsSession = this.connection.getSession(sessionId);
-      this.onDevtoolsAttached(devtoolsSession).catch(() => null);
+      if (this.onDevtoolsPanelOpened) this.onDevtoolsPanelOpened(devtoolsSession).catch(() => null);
```

Comment:
I don't quite follow this rename. Does this trigger everytime devtools opens if you close it?